lib-injection/sitecustomize.py

```diff
@@ -2,65 +2,253 @@
+        if os.path.exists(location):
+            with open(location, "r") as csvfile:
+                csv_reader = csv.reader(csvfile, delimiter=",")
+                for idx, row in enumerate(csv_reader):
+                    if idx < 2:
+                        continue
+                    min_pkgs[row[0]] = parse_version(row[1])
+            break
+    return min_pkgs
 
 
-debug_mode = os.environ.get("DD_TRACE_DEBUG", "").lower() in ("true", "1", "t")
-# Python versions that are supported by the current ddtrace release
-installable_py_versions = ("3.7", "3.8", "3.9", "3.10", "3.11", "3.12")
+def create_count_metric(metric, tags=None):
+    if tags is None:
+        tags = []
+    return {
+        "name": metric,
+        "tags": tags,
+    }
+
+
+def gen_telemetry_payload(telemetry_events):
+    return {
+        "metadata": {
+            "language_name": "python",
+            "language_version": PYTHON_VERSION,
+            "runtime_name": PYTHON_RUNTIME,
+            "runtime_version": PYTHON_VERSION,
+            "tracer_name": "python",
+            "tracer_version": INSTALLED_PACKAGES.get("ddtrace", "unknown"),
+            "pid": os.getpid(),
+        },
+        "points": telemetry_events,
+    }
+
+
+def send_telemetry(event):
+    event_json = json.dumps(event)
+    if not FORWARDER_EXECUTABLE or not TELEMETRY_ENABLED:
+        return
+    p = subprocess.Popen(
+        [FORWARDER_EXECUTABLE, str(os.getpid())],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    p.stdin.write(event_json)
+    p.stdin.close()
 
 
 def _get_clib():
     """Return the C library used by the system.
 
     If GNU is not detected then returns MUSL.
     """
-    import platform
 
     libc, version = platform.libc_ver()
     if libc == "glibc":
         return "gnu"
     return "musl"
 
 
-def _log(msg, *args, level="info"):
+def _log(msg, *args, **kwargs):
     """Log a message to stderr.
 
     This function is provided instead of built-in Python logging since we can't rely on any logger
     being configured.
     """
-    if debug_mode:
+    level = kwargs.get("level", "info")
+    if DEBUG_MODE:
         asctime = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
         msg = "[%s] [%s] datadog.autoinstrumentation(pid: %d): " % (asctime, level.upper(), os.getpid()) + msg % args
         print(msg, file=sys.stderr)
 
 
+def runtime_version_is_supported(python_runtime, python_version):
+    supported_versions = RUNTIMES_ALLOW_LIST.get(python_runtime, {})
+    if not supported_versions:
+        return False
+    return (
+        supported_versions["min"].version <= parse_version(python_version).version < supported_versions["max"].version
+    )
+
+
+def package_is_compatible(package_name, package_version):
+    installed_version = parse_version(package_version)
+    supported_version_spec = PKGS_ALLOW_LIST.get(package_name, Version((0,), ""))
```

Comment:
We should make the package names both in the list and the one we are checking against case insensitive. In my testing I saw an issue where package_name is `Flask` but in PKGS_ALLOW_LIST it's `flask`

Reply:
```suggestion
    supported_version_spec = PKGS_ALLOW_LIST.get(package_name.lower(), Version((0,), ""))
```

Reply:
Good call. I made a change ensuring that these comparisons are case-insensitive.